.github/workflows/release.yml

```diff
@@ -34,17 +35,62 @@ jobs:
           name: XenAPI
           path: scripts/examples/python/dist/
 
+  build-sdks:
+    name: Build and upload SDK artifacts
+    runs-on: ubuntu-latest
+    env:
+      XAPI_VERSION: ${{ github.ref_name }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup XenAPI environment
+        uses: ./.github/workflows/setup-xapi-environment
+        with:
+          xapi_version: ${{ env.XAPI_VERSION }}
+
+      - name: Generate and build SDKs
+        uses: ./.github/workflows/generate-and-build-sdks
+        with:
+          xapi_version: ${{ env.XAPI_VERSION }}
+
+      - name: Store C# SDK distribution artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: XenServer.NET
+          path: _build/install/default/xapi/sdk/csharp/src/bin/Release/*-prerelease-unsigned.nupkg
+
+      - name: Uninstall unversioned packages and remove pins
+        # This should purge them from the cache, unversioned package have
+        # 'master' as its version
+        run: |
+          opam list | awk -F " " '$2 == "master" { print $1 }' |  xargs opam uninstall
+          opam pin list | cut -f1 -d "." | xargs opam unpin
 
   release:
+    name: "Create and package release"
     runs-on: ubuntu-latest
-    needs: build-python
+    needs: [build-python, build-sdks]
     steps:
       - name: Retrieve python distribution artifacts
         uses: actions/download-artifact@v3
         with:
           name: XenAPI
           path: dist/
 
+      - name: Retrieve Python SDK distribution artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: XenAPI
+          path: dist/
```

Comment:
why is this step duplicated? Doesn't look right

Reply:
good point we might be able to make it a single call, I see some things in actions/download-artifact that I can try 👀

Reply:
just noticed you meant the fact that the python files are being downloaded twice, not that I'm calling the action twice! 🤦